hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java

```diff
@@ -242,4 +247,29 @@ public interface HederaOperations {
      * @param evmAddress    Evm address of hollow account
      */
     void externalizeHollowAccountMerge(@NonNull ContractID contractId, @Nullable Bytes evmAddress);
+
+    /**
+     * Given a {@link ContractID}, returns it if the shard and realm match for this node; otherwise,
+     * returns {@link #MISSING_CONTRACT_ID}.
+     *
+     * @param contractId the contract ID to validate
+     * @return the validated contract ID
+     */
+    ContractID shardAndRealmValidated(@NonNull ContractID contractId);
+
+    /**
+     * Given a {@link ContractID} and the current Hedera config, returns the given id if its shard and realm
+     * match the config; otherwise, returns {@link #MISSING_CONTRACT_ID}.
+     *
+     * @param contractId the contract ID to validate
+     * @param hederaConfig the current Hedera config
+     * @return the validated contract ID
+     */
+    default ContractID configValidated(@NonNull final ContractID contractId, @NonNull final HederaConfig hederaConfig) {
```

Comment:
not sure how this will affect the stuff I'm working on with allowing calls to non-existing contracts.  will test after this merges.

Reply:
Hmm good question...but I assume in any case we do not want to allow calls to contracts in a different shard?